x-pack/plugins/security_solution/public/overview/components/signals_by_category/use_filters_for_signals_by_category.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+
+import { Filter } from '../../../../../../../src/plugins/data/public';
+
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+import {
+  buildShowBuildingBlockFilter,
+  buildShowBuildingBlockFilterRuleRegistry,
+} from '../../../detections/components/alerts_table/default_config';
+
+// On the Overview page, in the Detection Alert Trend, we never show
+// "building block" alerts to remove noise from the Overview UI.
+// https://www.elastic.co/guide/en/security/current/building-block-rule.html
+const SHOW_BUILDING_BLOCK_ALERTS = false;
+
+export const useFiltersForSignalsByCategory = (baseFilters: Filter[] = []) => {
```

Comment:
I think that this type is incorrect here:
https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/overview/components/signals_by_category/index.tsx#L23

Where it is doing:

```ts
filters?: Filter[];
```

When really in that file it could be not optional like so:

```ts
filters: Filter[];
```


And then your code here can become:

```ts
export const useFiltersForSignalsByCategory = (baseFilters: Filter[]) => {
```

And you shouldn't need that defaulted into an array if it doesn't exist. I think it's that most programmers in other areas get used to marking everything as optional to avoid fixing tests or to reduce on TypeScript noise but it doesn't go well and increases other maintainers job of handling `undefined` when they don't need to. I would see if what I'm saying is right and if you could remove the optional setting within `signals_by_category`

Reply:
If you must keep this I would try using maybe a stable ref:

```ts
const stableRef: Filter[] = [];
export const useFiltersForSignalsByCategory = (baseFilters: Filter[] = stableRef) => {
```

Otherwise each time this gets called with an `undefined` you get a new re-render since it will always cause a new memo call.

Reply:
Yup, this makes sense 👍  I'll try to get rid of `?`, and if this doesn't work for some reason, use the constant array as the default value.